File: 2.两数相加/Jia.java
class Solution {
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
        int jin = 0;
        ListNode newList = new ListNode(0);
        ListNode temp1 = l1,temp2 = l2,temp3 = newList;
        while(temp1!=null||temp2!=null){
            temp3.next = new ListNode();
            temp3 = temp3.next;
            int a = temp1==null ?0:temp1.val;
            int b = temp2==null ?0:temp2.val;
            int now = a + b + jin;
            if(now>=10) {
                jin = 1;
                now -= 10;
            }else{
                jin = 0;
            }
            temp3.val = now;
            if(temp1!=null)
                temp1 = temp1.next;
            if(temp2!=null)
                temp2 = temp2.next;
        }
        if(jin == 1){
            temp3.next = new ListNode();
            temp3 = temp3.next;
            temp3.val = 1;
        }

        return newList.next;
    }
}

File: 2.两数相加/Jia.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */


struct ListNode* addTwoNumbers(struct ListNode* l1, struct ListNode* l2){
    struct ListNode *a,*out;
    int sum = 0,up = 0;
    a = (struct ListNode*)malloc(sizeof(struct ListNode));
    out = a;
    while(l1!=NULL&&l2!=NULL)
    {
        sum = l1->val + l2->val + up;
        if(sum>=10) {up = 1; sum-=10;} else {up = 0;}
        a->val= sum;
        if(l1->next!=NULL||l2->next!=NULL){
            a->next = (struct ListNode*)malloc(sizeof(struct ListNode));
            a = a->next;}
        a->next = NULL;
        l1 = l1->next;
        l2 = l2->next;
    }

    if(l1==NULL&&l2!=NULL)
    {
        while(l2!=NULL)
        {
            sum = l2->val + up;
            if(sum>=10) {up = 1; sum-=10;} else {up = 0;}
            a->val= sum;
            if(l2->next!=NULL){
            a->next = (struct ListNode*)malloc(sizeof(struct ListNode));
            a = a->next;}
            a->next = NULL;
            l2 = l2->next;
        }
    }
    if(l1!=NULL&&l2==NULL)
    {
        while(l1!=NULL)
        {
            sum = l1->val+ up;
            if(sum>=10) {up = 1; sum-=10;} else {up = 0;}
            a->val= sum;
            if(l1->next!=NULL){
            a->next = (struct ListNode*)malloc(sizeof(struct ListNode));
            a = a->next;}
            a->next = NULL;
            l1 = l1->next;
        }
    }
    if(up==1)
    {
        a->next = (struct ListNode*)malloc(sizeof(struct ListNode));
        a = a->next;
        a->val = up;
        a->next = NULL;
    }
    return out;
}
